app/controllers/hackdays_controller.rb

```diff
@@ -0,0 +1,77 @@
+class HackdaysController < ApplicationController
+
+  def index
+    @hackday = Hackday.new
+    set_current_and_past_hackdays
+  end
+
+  def create
+    @hackday = Hackday.new hackday_params
+    
+    if @hackday.save
+      cookies["#{@hackday.id}_votes"] = MAX_VOTES
+      redirect_to @hackday
+    else
+      set_current_and_past_hackdays
+      render :index
+    end
+  end
+
+  def edit
+    @hackday = Hackday.find(params[:id])
```

Comment:
1. find throws an exception. should do find_by_id which returns nil if not found.
2. should do params[:id].to_i. Otherwise it does a string look up which is worse.